sphinxext/rediraffe.py

```diff
@@ -320,20 +320,35 @@ def abs_path_in_src_dir_w_src_suffix(filename: str) -> Union[Path, None]:
                 logger.error(err_msg)
                 self.app.statuscode = 1
 
-        for renamed_file in rename_hints:
-            hint_to, perc = rename_hints[renamed_file]
-            if renamed_file in absolute_redirects:
-                logger.info(
-                    f"renamed file {renamed_file} redirects to {absolute_redirects[renamed_file]}."
-                )
-            else:
+        with redirects_path.open("a") as redirects_file:
+
+            for renamed_file in rename_hints:
+                hint_to, perc = rename_hints[renamed_file]
+
+                if renamed_file in absolute_redirects:
+                    logger.info(
+                        f"renamed file {renamed_file} redirects to {absolute_redirects[renamed_file]}."
+                    )
+                    continue
+
+                if self.name == "rediraffewritediff":
+                    if perc >= self.app.config.rediraffe_auto_redirect_perc:
+                        rel_rename_from = f'"{relpath(renamed_file, src_path)}"'
+                        rel_rename_to = f'"{relpath(hint_to, src_path)}"'
```

Comment:
```suggestion
                        rel_rename_from = f'"{str(PurePosixPath(renamed_file.relative_to(src_path)))}"'
                        rel_rename_to = f'"{str(PurePosixPath(hint_to.relative_to(src_path)))}"'
```
works on Windows. You need to add `PurePosixPath` to the pathlib imports.